cherab/core/laser/models/model.pyx

```diff
@@ -0,0 +1,145 @@
+from raysect.core.math.function.function3d import Constant3D
+from raysect.optical cimport Spectrum, Vector3D
+
+from cherab.core.math cimport ConstantVector3D
+from cherab.core cimport Plasma
+from cherab.core.laser.node cimport Laser
+from cherab.core.laser.models.model_base cimport LaserModel
+from cherab.core.laser.models.math_functions cimport AxisymmetricGaussian3D
+from libc.math cimport M_PI, sqrt, exp
+
+cdef class UniformPowerDensity(LaserModel):
+    def __init__(self, Laser laser = None, power = 1,  Vector3D polarization = Vector3D(0, 1, 0)):
+        super().__init__(laser)
+    
+        self.set_polarization(polarization)
+        self.set_pointing_function(ConstantVector3D(Vector3D(0, 0, 1)))
+        self.power = power
+
+    def set_polarization(self, Vector3D value):
+        value = value.normalise()
+        self.set_polarization_function(ConstantVector3D(value))
+
+    @property
+    def power(self):
+        return self._power
+
+    @power.setter
+    def power(self, double value):
+        if not value >0:
+            raise ValueError("Value has to be larger than 0.")
+        
+        self._power = value
+        self.set_power_density_function(Constant3D(value))
+
+cdef class GaussianBeamAxisymmetric(LaserModel):
+    def __init__(self, Laser laser = None, Vector3D polarization = Vector3D(0, 1, 0), power=1,
+                 laser_sigma = 0.01, waist_radius=0.001, m2 = 1, focus_z = 0.0, central_wavelength = 1064):
+
+        super().__init__(laser)
+
+        #laser sigma dependent constants
+        #set laser constants
+        self.polarization = polarization
+        self.waist_radius = waist_radius
+        self.m2 = m2
+        self.focus_z = focus_z
+
+    cpdef double get_power_axis(self, double z, double wavelength):
+        return self._power_const / self.get_beam_width2(z, wavelength) ** 2
+
+    cpdef Vector3D get_polarization(self, double x, double y, double z):
+        """
+        Returns vector denoting the laser polarisation.
+        
+        The point is specified in the laser beam space.
+        
+        :param x: x coordinate in meters.
+        :param y: y coordinate in meters. 
+        :param z: z coordinate in meters.
+        :return: power density in Wm^-3. 
+        """
+
+        return self._polarization_vector
+
+    cpdef double get_beam_width2(self, double z, double wavelength):
+        return self._waist2 + wavelength ** 2 * self._waist_const * (z - self._focus_z) ** 2
+
+    @property
+    def m2(self):
+        return self._m2
+    @m2.setter
+    def m2(self, value):
```

Comment:
what is m2?

Reply:
m2 (m^2) is a factor for deviation of a gaussan laser beam from an ideal case. It should be quite known constant in laser field, or at least I hit it a lot when I did some reading about Gaussian beam description.

Reply:
Is there not a better name for it? Can you come up with one? If not this will need at least some documented.

Reply:
I agree with @CnlPepper here. I'd rather this had a human-readable name.

Reply:
it is this: https://en.wikipedia.org/wiki/M_squared

I will rename it to beam_quality_factor, although I was told by our TS people and optician that nobody calls it like that and everybody uses m2. I will add to docstring that it is m^2